lib/evaluators/AsyncEvaluator.ts

```diff
@@ -26,7 +26,7 @@ export class AsyncEvaluator {
     return {
       now: context.now || new Date(Date.now()),
       baseIRI: context.baseIRI || undefined,
-      overloadCache: context.overloadCache || new LRUCache(),
+      overloadCache: {},
```

Comment:
I just realized an important thing.
I think we _do need_ to expose this field to the end-user, because we want this cache to be used Comunica-wide, across all actors.

So in Comunica, we would store this cache in the context, and pass this along to all actors that invoke sparqlee.

Reply:
That's a good point! How would you suggest solving this? (thinking about the change of types and how to avoid a major update)

Reply:
Either a different key, or a union type?

Reply:
I'd think a different key would be best? `fuctionArgumentsCache`?

Reply:
Sure, but maybe without the typo :P

Reply:
Not shure If we should change the name of the OverloadCache too. I feel like this name is most descriptive so internally it might be better to use this name still?

Reply:
Is there any place in comunica where multiple engines are created for the same data? If not keeping the cache might cause more harm to the communica code than that it is worth.

Reply:
> Not shure If we should change the name of the OverloadCache too. I feel like this name is most descriptive so internally it might be better to use this name still?

Sure, that's fine.

> Is there any place in comunica where multiple evaluators are created for the same data?

I see you already found it in the other PR.